sql: fork goyacc to handle more states

otan opened the pull request:
We've hit our limit of the number of states and actions in yacc, so
maintain a custom fork that bumps this limit.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/57384)
<!-- Reviewable:end -->

otan commented:
blocked by https://github.com/cockroachdb/cockroach/pull/57275

otan committed:
sql: fork goyacc to handle more states

We've hit our limit of the number of states and actions in yacc, so
maintain a custom fork that bumps this limit.

Release note: None

otan commented:
bors r=arulajmani

jordanlewis commented:
Yay! @otan I encourage you to send a PR (CL) to Go to up the states too - it was noncontroversial last time, and it would be good to upstream this change. :)
